plugin/interactive.go

```diff
@@ -22,13 +24,45 @@ func IsInteractive() bool {
 	return isInteractive
 }
 
+func tcGetpgrp(fd int) (pgrp int, err error) {
+	return unix.IoctlGetInt(fd, unix.TIOCGPGRP)
+}
+
+func tcSetpgrp(fd int, pgrp int) (err error) {
+	// Mimic IoctlSetPointerInt, which is not available on macOS.
+	v := int32(pgrp)
+	return unix.IoctlSetInt(fd, unix.TIOCSPGRP, int(uintptr(unsafe.Pointer(&v))))
+}
+
 // Prompt prints the supplied message, then waits for input on stdin.
 func Prompt(msg string) (string, error) {
 	if !IsInteractive() {
-		return "", fmt.Errorf("Not an interactive session")
+		return "", fmt.Errorf("not an interactive session")
 	}
-	fmt.Fprintf(os.Stderr, "%s: ", msg)
+
+	inFd := int(os.Stdin.Fd())
```

Comment:
I think a comment giving a high-level overview of this logic would help (e.g. precise meaning of "We control Stdin" and "If we don't control STDIN, here's how we get back that control temporarily").

Reply:
Much clearer. I still think it'd be useful to explicitly state the pgrp connection (e.g. "control of stdin" means "stdin's pgrp is our pgrp" and "temporarily controlling stdin" means "temporarily setting its pgrp to our pgrp") but that's up to you.

Reply:
I think I'm going to leave it as-is. I don't feel like stating `stdin's pgrp is our pgrp` provides anything that's not explained by reading the code.